src/sage/graphs/generic_graph.py

```diff
@@ -938,6 +939,143 @@ def _latex_(self):
 
         return self.latex_options().latex()
 
+    def tikz(self, format='dot2tex', edge_labels=None,
+            color_by_label=False, prog='dot', rankdir='down',
+            use_sage_preamble=None, **kwds):
+        r"""
+        Return a TikzPicture of the graph.
+
+        INPUT:
+
+        - ``format`` -- string (default: ``None``), ``'dot2tex'`` or
+          ``'tkz_graph'``. If ``None``, it is set to ``'dot2tex'`` if
+          dot2tex is present, otherwise it is set to ``'tkz_graph'``.
+        - ``edge_labels`` -- bool (default: ``None``), if ``None``
+          it is set to ``True`` if and only if format is ``'dot2tex'``
+        - ``color_by_label`` -- bool (default: ``False``)
```

Comment:
could you elaborate on what happens when this is `True` ?

Reply:
Done